File: Solutions/The-Clockwork-Town-of-Tempora.js
function timeDifference(clockTime, grandClockTime) {
    const [clockHour, clockMinute] = clockTime.split(":").map(Number);
    const [grandClockHour, grandClockMinute] = grandClockTime.split(":").map(Number);

    return (clockHour - grandClockHour) * 60 + (clockMinute - grandClockMinute);
}

function synchronizeClocks(clockTimes, grandClockTime) {
    return clockTimes.map(clockTime => timeDifference(clockTime, grandClockTime));
}

const clockTimes = ["14:45", "15:05", "15:00", "14:40"];
const grandClockTime = "15:00";
console.log(synchronizeClocks(clockTimes, grandClockTime));  // [-15, 5, 0, -20]

File: Solutions/Python/The-Clockwork-Town-of-Tempora.py
def time_difference(clock_time, grand_clock_time):
    clock_hour, clock_minute = map(int, clock_time.split(":"))
    grand_clock_hour, grand_clock_minute = map(int, grand_clock_time.split(":"))

    return (clock_hour - grand_clock_hour) * 60 + (clock_minute - grand_clock_minute)

def synchronize_clocks(clock_times, grand_clock_time):
    return [time_difference(clock_time, grand_clock_time) for clock_time in clock_times]

clock_times = ["14:45", "15:05", "15:00", "14:40"]
grand_clock_time = "15:00"
print(synchronize_clocks(clock_times, grand_clock_time))  # [-15, 5, 0, -20]